### PR TITLE
Refactor/class inheritance

### LINKED ISSUE #27: A parent Fitness class
As a dev

I would like to have a parent class that have common functions such as "getAverage()"

so i can extend it to other subclasses and use parent functions when i need to get general information, such as

"average values", or filtered data by Id

### PR DESCRIPTION
**Pull request by cameronRomo**
## What is the change?
* Refactor class inheritance by adding a parent class called Fitness

## What does it fix? 
* Classes are now more dynamic and use inheritance
* These changes were respectively implemented in `scripts.js`

## Is this a fix or a feature?  
* This is a feature

## Where should the reviewer start?
*`Activity.js` and test 
  * lines 3 - 37
* `Fitness.js` and test
  * lines 1 - 22
* `Hydration.js` and test
  * lines 3 - 15
* `Sleep.js` and test
  * lines 4 - 77
* `User-repo.js` and test
  * lines  1 - 81
* `scripts.js`
  * lines 216 & 255

## How should this be tested?
* This can be tested by pulling down and running `npm test`

### CODE REVIEW

File: src/Fitness.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Fitness {
```
Comment (okalimbet) [inline src/Fitness.js:1]: Thank you for creating a Fitness class following our instruction guidelines regarding the inheritance. glad we could refactor some repetitive functions

**Review by okalimbet — APPROVED** [review]
Nice replacement of child functions with parent extension !

File: src/Activity.js [hunk 2]
```diff
@@ -33,9 +33,8 @@ class Activity extends Fitness {
       .map(data => data.date);
   }
 
-  getStairRecord(id) {
-    return this.activityData
-      .filter(data => id === data.userID)
+  getStairRecord(id, dataType) {
+    return this.filterDataByUserId(id, dataType)
```
Comment (okalimbet) [inline src/Activity.js:37]: missing semicolon line 37

File: src/User-repo.js [hunk 3]
```diff
@@ -11,14 +11,11 @@ class UserRepo extends Fitness {
...
+    return this.filterDataByUserId(id, dataSet);
   }
 
-  calculateAverageStepGoal() {
-    var totalStepGoal = this.users.reduce((total, data) => {
-      return total = total + data.dailyStepGoal;
-    }, 0);
-    return totalStepGoal / this.users.length;
+  calculateAverageStepGoal(data, property) {
+    return this.getAverage(data, property)
```
Comment (okalimbet) [inline src/User-repo.js:18]: missing semicolon line 18

File: test/Fitness-test.js [hunk 4]
```diff
@@ -346,4 +346,160 @@ describe('Fitness', function() {
       flightsOfStairs: 16
```
Comment (okalimbet) [inline test/Fitness-test.js:346]: we would need reorganize Fitness-test.js file with beforeEach to include all data  at the top of the test file if we get time in the end:)

**Review by okalimbet — APPROVED** [review]

File: src/Activity.js [hunk 5]
```diff
@@ -1,19 +1,18 @@
-class Activity {
-  constructor(activityData) {
-    this.activityData = activityData
-  }
+import Fitness from './Fitness';
 
-  getUserInfoByDateAndId(id, date) {
-    return this.activityData.find(data => id === data.userID && date === data.date);
+class Activity extends Fitness {
```
Comment (mdflynn) [inline src/Activity.js:3]: Great job using object inheritance!

File: src/Fitness.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  constructor(data) {
+    this.data = data;
+  }
+
+  getAverage(data, property = null) {
+    return data.reduce((total, value) => {
+      property === null ? total += value : total += value[property]
```
Comment (mdflynn) [inline src/Fitness.js:8]: semi-colon at the end of this line.

**Review by mdflynn — APPROVED** [review]
This is a nice addition to the structure of the project.

**Pull request merged**